extensions/kubernetes/vanilla/runtime/src/main/java/io/quarkus/kubernetes/runtime/config/KubernetesConfigFallback.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.kubernetes.runtime.config;
+
+import java.util.function.Function;
+
+import io.smallrye.config.FallbackConfigSourceInterceptor;
+
+public class KubernetesConfigFallback extends FallbackConfigSourceInterceptor {
+    public KubernetesConfigFallback() {
+        super(new Function<String, String>() {
+            @Override
+            public String apply(final String name) {
+                if (name.startsWith("quarkus.openshift.")) {
+                    return "quarkus.kubernetes." + name.substring(18);
+                } else if (name.startsWith("quarkus.knative.")) {
+                    return "quarkus.kubernetes." + name.substring(16);
```

Comment:
Please, use constants for `quarkus.kubernetes.`, `quarkus.knative.` and `quarkus.openshift.` and avoid the magic numbers here.